src/edge.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import Optional
+
+import paramiko
+from scp import SCPClient
+
+
+class EdgeConnection(object):
+    def __init__(self, address: str, username: str, certificate_path: str, port: int = 22):
+        self.address = address
+        self.username = username
+        self.certificate = certificate_path
+        self.port = port
+        self.ssh_client = None
+        self.ssh_client = self.connect()
+
+    def connect(self) -> Optional[paramiko.SSHClient]:
+        try:
+            ssh_client = paramiko.SSHClient()
+            ssh_client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            ssh_client.connect(
+                hostname=self.address,
+                port=self.port,
+                username=self.username,
+                key_filename=self.certificate,
+                timeout=20,
+                look_for_keys=False,
+            )
+            logging.info(f"Connected to {self.address} over SSH")
+
+        except Exception as e:
+            ssh_client = None
+            logging.error(f"Unable to connect to {self.address} over SSH: {e}")
```

Comment:
Isn't it easier to raise exception here and later in the code to assume that `ssh_client` is not `Optional`? I'm not sure about the value of letting tool continue if this step fails